module/zfs/zfs_fm.c

```diff
@@ -1444,6 +1444,48 @@ zfs_ereport_fini(void)
 	mutex_destroy(&recent_events_lock);
 }
 
+void
+zfs_ereport_snapshot_post(const char *subclass, spa_t *spa, const char *name)
+{
+	nvlist_t *aux;
+
+	aux = fm_nvlist_create(NULL);
+	nvlist_add_string(aux, "snapshot_name", name);
+
+	zfs_post_common(spa, NULL, FM_RSRC_CLASS, subclass, aux);
+	fm_nvlist_destroy(aux, FM_NVA_FREE);
+}
+
+void
+zfs_ereport_zvol_post(const char *subclass, const char *name, const char *bsd,
+    const char *rbsd)
```

Comment:
I assume `bsd` and `rbsd` only for BSD?  If so, what should Linux pass for these values?

Reply:
Right, it could be in `#ifdef _APPLE_` as I did not imagine others to consume it. Although, I do use it in Windows as well, where I send the logic path, and raw path. Could be renamed to `path` and `physpath` to fit better with existing code. They are currently just passed to zedlets to do whatever is needed.
(In our case, we create the symlinks for users to be able to locate each zvol).

This brings up a larger discussion we've had at macOS port. When you create a ZVOL, say `dataset/vol`, we create, for example, `/dev/disk3` for it.
There is no way for a user to connect the two. (for whatever purpose, formatting, /etc/fstab etc). Which is why we create the symlink `/var/run/zfs/dsk/pool/dataset/vol -> /dev/disk3`. (Can't create dirs in /dev)

What would be nice is a `zfs get devname dataset/vol` which returns `/dev/disk3`. But we've avoided polluting the dataset properties as much as possible - unless of course - that would be useful to other platforms?

Reply:
So using your example, would the variables be: `bsd = /var/run/zfs/dsk/pool/dataset/vol` and `rbsd = /dev/disk3`? 

If so, yea, we may want to rename these variables and their nvlist names so we could use them on all platforms (or leave them as NULL if not applicable).

Reply:
What do you think about renaming the arguments and adding this comment?:
```
/*
 * Post when a event when a zvol is created or removed
 *
 * This is currently only used by OSX, since it uses the event to cue creating
 * symlinks between the volume name (mypool/myvol) and the actual /dev
 * device (/dev/disk3).  For example:
 *
 * /var/run/zfs/dsk/mypool/myvol -> /dev/disk3
 *
 * name: The full name of the zvol ("mypool/myvol")
 * dev_name: The full /dev name for the zvol ("/dev/disk3")
 */
```